spec/index.bs

```diff
@@ -2510,146 +2504,227 @@ Issue(https://github.com/gpuweb/gpuweb/issues/851): consider making `textureComp
+        <td rowspan=3>{{GPUTextureView}}
+        <td>{{GPUTextureType/"filterable"}}
         <td>[=internal usage/constant=]
     <tr>
-        <td>{{GPUBindingType/"readonly-storage-texture"}}
-        <td>{{GPUTextureView}}
+        <td>{{GPUTextureType/"unfilterable"}}
+        <td>[=internal usage/constant=]
+    <tr>
+        <td>{{GPUTextureType/"multisample"}}
+        <td>[=internal usage/constant=]
+
+    <tr>
+        <td rowspan=2>{{GPUBindGroupLayoutEntry/storageTexture}}
+        <td rowspan=2>{{GPUTextureView}}
+        <td>{{GPUStorageTextureType/"readonly"}}
         <td>[=internal usage/storage-read=]
     <tr>
-        <td>{{GPUBindingType/"writeonly-storage-texture"}}
-        <td>{{GPUTextureView}}
+        <td>{{GPUStorageTextureType/"writeonly"}}
         <td>[=internal usage/storage-write=]
 </table>
 
+<div algorithm>
+    To get the <dfn abstract-op>layout entry binding type</dfn> in a given {{GPUBindGroupLayoutEntry}} |entry|:
+
+    1. If |entry|.{{GPUBindGroupLayoutEntry/buffer}} is not `undefined`:
+        1. Return |entry|.{{GPUBindGroupLayoutEntry/buffer}}.{{GPUBufferLayoutEntry/type}}.
+    1. If |entry|.{{GPUBindGroupLayoutEntry/sampler}} is not `undefined`:
+        1. Return |entry|.{{GPUBindGroupLayoutEntry/sampler}}.{{GPUSamplerLayoutEntry/type}}.
+    1. If |entry|.{{GPUBindGroupLayoutEntry/texture}} is not `undefined`:
+        1. Return |entry|.{{GPUBindGroupLayoutEntry/texture}}.{{GPUTextureLayoutEntry/type}}.
+    1. If |entry|.{{GPUBindGroupLayoutEntry/storageTexture}} is not `undefined`:
+        1. Return |entry|.{{GPUBindGroupLayoutEntry/storageTexture}}.{{GPUStorageTextureLayoutEntry/type}}.
+</div>
+
+<script type=idl>
+enum GPUBufferType {
+    "uniform",
+    "storage",
+    "readonly-storage",
+};
+
+dictionary GPUBufferLayoutEntry {
+    GPUBufferType type = "uniform";
+    boolean hasDynamicOffset = false;
+    GPUSize64 minBufferBindingSize = 0;
+};
+</script>
+
+{{GPUBufferLayoutEntry}} dictionaries have the following members:
+
+<dl dfn-type=dict-member dfn-for=GPUBufferLayoutEntry>
+    : <dfn>type</dfn>
+    ::
+        Indicates the type required for buffers bound to this bindings.
+
+    : <dfn>hasDynamicOffset</dfn>
+    ::
+        Indicates whether this binding requires a dynamic offset.
+
+    : <dfn>minBufferBindingSize</dfn>
+    ::
+        May be used to indicate the minimum buffer binding size.
+</dl>
+
+<script type=idl>
+enum GPUSamplerType {
+    "filtering",
+    "non-filtering",
+    "comparison",
+};
+
+dictionary GPUSamplerLayoutEntry {
+    GPUSamplerType type = "filtering";
+};
+</script>
+
+{{GPUSamplerLayoutEntry}} dictionaries have the following members:
+
+<dl dfn-type=dict-member dfn-for=GPUSamplerLayoutEntry>
+    : <dfn>type</dfn>
+    ::
+        Indicates the required type of a sampler bound to this bindings.
+</dl>
+
+<script type=idl>
+enum GPUTextureType {
+    "filterable",
+    "unfilterable",
+    "multisample",
```

Comment:
idea: since they others describe what can be done ("-able") maybe this should be the same? e.g. "sample-addressable" or "multisample-addressable"